src/features/edit/reconnect.ts

```diff
@@ -0,0 +1,98 @@
+/********************************************************************************
+ * Copyright (c) 2017-2018 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { Routable, isRoutable } from "./model";
+import { SModelElement } from "../../base/model/smodel";
+import { Action } from "../../base/actions/action";
+import { Command, CommandExecutionContext, CommandResult } from "../../base/commands/command";
+
+export const connectableFeature = Symbol('connectableFeature');
+
+export interface Connectable {
+    canConnect(routable: Routable, role: 'source' | 'target'): boolean;
+}
+
+export function isConnectable<T extends SModelElement>(element: T): element is Connectable & T {
```

Comment:
Should this go to `model.ts`?

Reply:
We have so many different functionalities here I decided to separate them into separate files.